numba/np/arrayobj.py

```diff
@@ -1970,16 +1970,66 @@ def imp(ary):
     return res
 
 
+@register_jitable
+def _np_clip_impl(a, a_min, a_max, out):
+    # Both a_min and a_max are numpy arrays
+    ret = np.empty_like(a) if out is None else out
+    a_b, a_min_b, a_max_b = np.broadcast_arrays(a, a_min, a_max)
+    for index in np.ndindex(a_b.shape):
+        val_a = a_b[index]
+        val_a_min = a_min_b[index]
+        val_a_max = a_max_b[index]
+        ret[index] = min(max(val_a, val_a_min), val_a_max)
+
+    return ret
+
+
+@register_jitable
+def _np_clip_impl_none1(a, a_min, a_max, out):
```

Comment:
Rename this function to something that reflects that `a_min` is None here.
```suggestion
def _np_clip_impl_amin_none(a, a_min, a_max, out):
```